app/src/main/java/org/thoughtcrime/securesms/mediasend/CameraButtonView.java

```diff
@@ -0,0 +1,302 @@
+  }
+
+  @Override
+  public void setOnLongClickListener(@Nullable OnLongClickListener listener) {
+    throw new IllegalStateException("Use setVideoCaptureListener instead");
+  }
+
+  public void setVideoCaptureListener(@Nullable VideoCaptureListener videoCaptureListener) {
+    if (isRecordingVideo) throw new IllegalStateException("Cannot set video capture listener while recording");
+
+    if (videoCaptureListener != null) {
+      this.cameraButtonMode = CameraButtonMode.MIXED;
+      this.videoCaptureListener = videoCaptureListener;
+      super.setOnLongClickListener(internalLongClickListener);
+    } else {
+      this.cameraButtonMode = CameraButtonMode.IMAGE;
+      this.videoCaptureListener = null;
+      super.setOnLongClickListener(null);
+    }
+  }
+
+  public void setProgress(float percentage) {
+    progressPercent = Util.clamp(percentage, 0f, 1f);
+    invalidate();
+  }
+
+  @Override
+  public boolean onTouchEvent(MotionEvent event) {
+    if (cameraButtonMode == CameraButtonMode.IMAGE) {
+      return handleImageModeTouchEvent(event);
+    }
+
+    boolean eventWasHandled = handleVideoModeTouchEvent(event);
+    int     action          = event.getAction();
+
+    if (action == MotionEvent.ACTION_UP || action == MotionEvent.ACTION_CANCEL) {
+      isRecordingVideo = false;
+    }
+
+    return eventWasHandled;
+  }
+
+  @Override
+  protected void onLayout(boolean changed, int left, int top, int right, int bottom) {
+    super.onLayout(changed, left, top, right, bottom);
+
+    getLocalVisibleRect(deadzoneRect);
+    deadzoneRect.left   += (int) (getWidth()  * DEADZONE_REDUCTION_PERCENT / 2f);
+    deadzoneRect.top    += (int) (getHeight() * DEADZONE_REDUCTION_PERCENT / 2f);
+    deadzoneRect.right  -= (int) (getWidth()  * DEADZONE_REDUCTION_PERCENT / 2f);
+    deadzoneRect.bottom -= (int) (getHeight() * DEADZONE_REDUCTION_PERCENT / 2f);
+  }
+
+  private boolean handleImageModeTouchEvent(MotionEvent event) {
+    int action = event.getAction();
+    switch (action) {
+      case MotionEvent.ACTION_DOWN:
+        if (isEnabled()) {
+          startAnimation(shrinkAnimation);
+          performClick();
+        }
+        return true;
+      case MotionEvent.ACTION_UP:
+        startAnimation(growAnimation);
+        return true;
+      default:
+        return super.onTouchEvent(event);
+    }
+  }
+
+  private boolean handleVideoModeTouchEvent(MotionEvent event) {
+    int action = event.getAction();
+    switch (action) {
+      case MotionEvent.ACTION_DOWN:
+        if (isEnabled()) {
+          startAnimation(shrinkAnimation);
+        }
+      case MotionEvent.ACTION_MOVE:
+        if (isRecordingVideo && eventIsNotInsideDeadzone(event)) {
+
+          float maxRange  = getHeight() * DRAG_DISTANCE_MULTIPLIER;
+          float deltaY    = Math.abs(event.getY() - deadzoneRect.top);
+          float increment = Math.min(1f, deltaY / maxRange);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The result of `Integer` multiplication is cast into a larger type. If you think the result of the multiplication could exceed the size of an `Integer`, consider casting the arguments of the multiplication to avoid arithmetic overflows. For more information, see [CWE-190](https://cwe.mitre.org/data/definitions/190.html).